nestkernel/per_thread_bool_indicator.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  per_thread_bool_indicator.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "per_thread_bool_indicator.h"
+
+namespace nest
+{
+
+BoolIndicatorUInt64::BoolIndicatorUInt64()
+  : status_( false_uint64 )
+{
+}
+
+BoolIndicatorUInt64::BoolIndicatorUInt64( const bool status )
+  : status_( status )
+{
+}
+
+bool
+BoolIndicatorUInt64::is_true() const
+{
+  return ( status_ == true_uint64 );
+}
+
+bool
+BoolIndicatorUInt64::is_false() const
+{
+  return ( status_ == false_uint64 );
+}
+
+void
+BoolIndicatorUInt64::set_true()
+{
+  status_ = true_uint64;
+}
+
+void
+BoolIndicatorUInt64::set_false()
+{
+  status_ = false_uint64;
+}
+
+void
+BoolIndicatorUInt64::logical_and( const bool status )
+{
+  status_ = ( static_cast< bool >( status_ ) and status );
```

Comment:
maybe something like this? but i don't feel too strongly about it

```suggestion
  if ( static_cast< bool >( status_ ) and status )
  {
      set_true();
  }
  else
  {
      set_false();
  }
```

Reply:
I find the one line version more clear and the static_cast<> expresses that a sequence of conversions is going on.

Reply:
I have not used an `if/else`-solution as I think the one-line version is unambiguous. It is well defined by the C++ International Standard what happens when assigning a `bool` value to an integer variable.

Reply:
I agree with @suku248. The `if-else` is indeed to complex. If one wanted to be explicit, one could use
```c++
status_ = ( static_cast< bool >( status_ ) and status ) ? true_uint64 : false_uint64;
```
but I think that confuses more than it clarifies.